apps/studio/prisma/schema.prisma

```diff
@@ -26,68 +26,73 @@ model VerificationToken {
   expires    DateTime
 }
 
-model Post {
-  id           String       @id @default(cuid())
-  title        String?      @db.Text
-  content      String       @db.Text
-  contentHtml  String       @db.Text
-  authorId     String
-  author       User         @relation(fields: [authorId], references: [id], onDelete: Cascade)
-  images       String[]
-  likes        LikedPosts[]
-  // This features the self-relation Prisma concept, specifically a one-to-many self relation.
-  // https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations
-  parentPostId String?
-  // The parent post of this post, if it is a reply.
-  // @relation("PostToParent") is the name of the relation. See relation documentation below for more info.
-  parent       Post?        @relation("PostReplies", fields: [parentPostId], references: [id])
-  // The replies to this post.
-  replies      Post[]       @relation("PostReplies")
-
-  // To return `Date`s intact through the API we use transformers
-  // https://trpc.io/docs/data-transformers
-  // This is unique so it can be used for cursor-based pagination
-  createdAt DateTime  @default(now())
-  updatedAt DateTime  @default(now()) @updatedAt
-  deletedAt DateTime? @db.Timestamp
-
-  @@index([parentPostId])
-  @@index([parentPostId, deletedAt, id])
-  @@index([deletedAt])
-  @@index([createdAt])
-}
+model Resource {
+  id             String       @id @default(cuid())
+  name           String
+  siteId         String
+  parentId       String?
+  lastEditTime   DateTime?
+  lastEditUserId Int
+  blobId         String?      @unique
+  variant        ResourceType
+  userId         String
 
-model LikedPosts {
-  post      Post     @relation(fields: [postId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  postId    String
-  user      User     @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  userId    String
-  createdAt DateTime @default(now())
+  site         Site         @relation(fields: [siteId], references: [id])
+  parent       Resource?    @relation("parentRelation", fields: [parentId], references: [id])
+  child        Resource[]   @relation("parentRelation")
+  lastEditUser User?        @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction) // Don't delete the file if the user is deleted
+  blob         Blob?
```

Comment:
Hmm we might not want to store the blob directly in the resource table! We only really need to retrieve one file's content at a time, and this might end up bloating the table we use primarily to retrieve the structure of the pages

Reply:
ah oops, should be blobId?